llvm/lib/TargetParser/Triple.cpp

```diff
@@ -1488,6 +1488,48 @@ VersionTuple Triple::getVulkanVersion() const {
   return VersionTuple(0);
 }
 
+VersionTuple Triple::getDXILVersion() const {
+  if (getArch() != dxil || getOS() != ShaderModel)
+    llvm_unreachable("invalid DXIL triple");
+
+  VersionTuple DXILVersion;
+  switch (getSubArch()) {
+  case Triple::NoSubArch:
+  case Triple::DXILSubArch_v1_0:
+    DXILVersion = VersionTuple(1, 0);
```

Comment:
Can this be simplified using ParseVersionFromName?

Reply:
Updated.